draft_python/mtg_draft_ai/controller.py

```diff
@@ -0,0 +1,128 @@
+import random
+from mtg_draft_ai.api import *
+
+
+class DraftController:
+    """Runs a draft by asking each drafter to pick from the right pack in the right order."""
+
+    def __init__(self, draft_info, drafters, packs):
+        """Basic init method which does no manipulation of its parameters.
+
+        For most cases, use DraftController.create instead, which creates shuffled packs
+        from the card list.
+
+        Args:
+            draft_info (api.DraftInfo): Config about the draft, which will not change
+                during the draft.
+            drafters (list of api.Drafter): The Drafters, which may have different Picker
+                implementations.
+            packs (api.Packs): The Packs to use for this draft. Should already be initialized.
+        """
+        self.draft_info = draft_info
+        self.drafters = drafters
+        self.packs = packs
+
+    @staticmethod
+    def create(draft_info, drafters):
+        """Creates a DraftController with shuffled packs generated from the card list.
+
+        Args:
+            draft_info (api.DraftInfo): Config about the draft, which will not change
+                during the draft.
+            drafters (list of api.Drafter): The Drafters, which may have different Picker
+                implementations.
+        Returns:
+            DraftController: A DraftController initialized with shuffled packs.
+        """
+        if draft_info.num_drafters != len(drafters):
```

Comment:
Hmm, I made the DraftInfo class initially to have one object to pass in to the Drafter/Picker which contains the information about the draft configuration. E.g. a Picker might need to know number of drafters to determine if it can wheel a card.

Agreed it feels a little awkward here, not sure how to improve it though, any ideas?

Reply:
Hmm. Theoretically, we could instantiate the drafters, add them to the list, and give them the size of the table after they've all 'sat down'. But it seems like overkill to me. This should be fine.

Reply:
I agree this looks a bit awkward, but I think I like having the separate variable dictate what the lengths 'should' be.